bundler/lib/dependabot/bundler/update_checker/latest_version_finder/dependency_source.rb

```diff
@@ -47,27 +47,32 @@ def versions
           def latest_git_version_details
             return unless git?
 
-            dependency_source_details =
+            source_details =
               dependency.requirements.map { |r| r.fetch(:source) }.
               uniq.compact.first
 
-            in_a_temporary_bundler_context do
-              SharedHelpers.with_git_configured(credentials: credentials) do
-                # Note: we don't set `ref`, as we want to unpin the dependency
-                source = ::Bundler::Source::Git.new(
-                  "uri" => dependency_source_details[:url],
-                  "branch" => dependency_source_details[:branch],
-                  "name" => dependency.name,
-                  "submodules" => true
+            SharedHelpers.with_git_configured(credentials: credentials) do
+              in_a_native_bundler_context do |tmp_dir|
+                SharedHelpers.run_helper_subprocess(
+                  command: NativeHelpers.helper_path,
+                  function: "depencency_source_latest_git_version",
+                  args: {
+                    dir: tmp_dir,
+                    gemfile_name: gemfile.name,
+                    dependency_name: dependency.name,
+                    credentials: credentials,
+                    dependency_source_url: source_details[:url],
+                    dependency_source_branch: source_details[:branch]
+                  }
                 )
-
-                # Tell Bundler we're fine with fetching the source remotely
-                source.instance_variable_set(:@allow_remote, true)
-
-                spec = source.specs.first
-                { version: spec.version, commit_sha: spec.source.revision }
               end
+            end.transform_keys(&:to_sym)
+          rescue Dependabot::SharedHelpers::HelperSubprocessFailed => e
```

Comment:
TODO: This should be refactored to align with the subprocess error handling in `VersionResolver`

Reply:
👍  yeah was running into some issues here were `error_class` wasn't set